hw17/Vector2d.cpp

```diff
@@ -0,0 +1,88 @@
+#include "Vector2d.h"
+#include <iostream>
+
+int Vector2d::activeInstancesCount = 0;
+
+Vector2d::Vector2d() : m_x(0.0f), m_y(0.0f) {
+    activeInstancesCount++;
+}
+
+Vector2d::Vector2d(float x, float y) : m_x(x), m_y(y) {
+    activeInstancesCount++;
+}
+
+Vector2d::~Vector2d() {
+    activeInstancesCount--;
+}
+
+Vector2d& Vector2d::operator=(const Vector2d& other) {
+    if (this != &other) {
+        m_x = other.m_x;
+        m_y = other.m_y;
+    }
+    return *this;
+}
+
+Vector2d Vector2d::operator+(const Vector2d& other) const {
+    return Vector2d(m_x + other.m_x, m_y + other.m_y);
+}
+
+Vector2d Vector2d::operator-(const Vector2d& secondVector) const {
+    return Vector2d(m_x - secondVector.m_x, m_y - secondVector.m_y);
+}
+
+float Vector2d::operator()() const {
+    return std::sqrt(m_x * m_x + m_y * m_y);
+}
+
+float& Vector2d::operator[](int i) {
+    if (i == 0) {
+        return m_x;
+    } else if (i == 1) {
+        return m_y;
+    } else {
+        return m_x;
+    }
```

Comment:
assert or throw an exception

Reply:
I wasn't sure if it was possible to use exceptions